### PR TITLE
add new API paddle.linalg.lu/lu_unpack

### PR DESCRIPTION
**Pull request by zhiboniu**
### PR types

New features
### PR changes

APIs
### Describe

add new API paddle.linalg.lu/lu_unpack

            import paddle 

            x = paddle.to_tensor([[1.0, 2.0], [3.0, 4.0], [5.0, 6.0]]).astype('float64')
            lu,p,info = paddle.linalg.lu(x, get_infos=True)

            '''
            >>> lu:
            Tensor(shape=[3, 2], dtype=float64, place=CUDAPlace(0), stop_gradient=True,
                [[5.        , 6.        ],
                    [0.20000000, 0.80000000],
                    [0.60000000, 0.50000000]])
            >>> p
            Tensor(shape=[2], dtype=int32, place=CUDAPlace(0), stop_gradient=True,
                [3, 3])
            >>> info
            Tensor(shape=[], dtype=int32, place=CUDAPlace(0), stop_gradient=True,
                0)
            '''

            P,L,U = paddle.linalg.lu_unpack(lu,p)

            '''
            >>> P
            (Tensor(shape=[3, 3], dtype=float64, place=CUDAPlace(0), stop_gradient=True,
            [[0., 1., 0.],
            [0., 0., 1.],
            [1., 0., 0.]]), 
            >>> L
            Tensor(shape=[3, 2], dtype=float64, place=CUDAPlace(0), stop_gradient=True,
            [[1.        , 0.        ],
            [0.20000000, 1.        ],
            [0.60000000, 0.50000000]]), 
            >>> U
            Tensor(shape=[2, 2], dtype=float64, place=CUDAPlace(0), stop_gradient=True,
            [[5.        , 6.        ],
            [0.        , 0.80000000]]))
            '''

image
image

image

### CODE REVIEW

File: python/paddle/tensor/linalg.py [hunk 1]
```diff
@@ -1823,6 +1823,205 @@ def qr(x, mode="reduced", name=None):
...
 
+def lu(x, pivot=True, get_infos=False, name=None):
+    r"""
+    Computes the LU factorization of an N-D(N>=2) matrix x. 
+
+    Returns the LU factorization(inplace x) and Pivots. low triangular matrix L and 
+    upper triangular matrix U are combined to a single LU matrix.
+
+    Pivoting is done if pivot is set to True.
+    P mat can be get by pivots:
```
Comment (jzhang533) [inline python/paddle/tensor/linalg.py:1834]: 这段代码是想说明怎么用返回的`pivots`吧，是不是挪到示例代码里比较好。

File: python/paddle/tensor/linalg.py [hunk 2]
```diff
@@ -1823,6 +1823,205 @@ def qr(x, mode="reduced", name=None):
...
+    # ones = eye(rows) #eye matrix of rank rows
+    # for i in range(cols):
+    #     swap(ones[i], ones[pivots[i]])
+    # return ones
+
+    Args:
+
+        X (Tensor): the tensor to factor of N-dimensions(N>=2).
```
Comment (jzhang533) [inline python/paddle/tensor/linalg.py:1842]: 小写`x`

File: python/paddle/tensor/linalg.py [hunk 3]
```diff
@@ -1823,6 +1823,205 @@ def qr(x, mode="reduced", name=None):
...
+
+        get_infos (bool, optional): if set to True, returns an info IntTensor. Default: False.
+
+        name (str, optional): Name for the operation (optional, default is None).
+            For more information, please refer to :ref:`api_guide_Name`.
+            
+    Returns:
+        factorization (Tensor): LU matrix, the factorization of input X.
+
+        pivots (IntTensor): the pivots of size(∗(N-2), min(m,n)). `pivots` stores all the
```
Comment (jzhang533) [inline python/paddle/tensor/linalg.py:1854]: paddle里有`IntTensor`吗？ 还是想说返回的是`Tensor with dtype=int`？

File: python/paddle/tensor/linalg.py (hunk 1 above)
Comment (zhiboniu) [reply python/paddle/tensor/linalg.py:1834]: 嗯，这是一段伪代码，纯粹说明转换关系的作用

File: python/paddle/tensor/linalg.py (hunk 3 above)
Comment (zhiboniu) [reply python/paddle/tensor/linalg.py:1854]: 是想说明Tensor with dtype=int

**Review by jzhang533 — APPROVED** [review]
will update docs in following PR

File: python/paddle/tensor/linalg.py (hunk 2 above)
Comment (zhiboniu) [reply python/paddle/tensor/linalg.py:1842]: 考虑到ci执行时间太久，今天着急合入，文档修改单独提pr或者后面pr修改。

**Review by jeff41404 — APPROVED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**